packages/runtime/src/runtime/history.tsx

```diff
@@ -0,0 +1,11 @@
+import { type NavigateFunction, type Location } from "react-router-dom";
+
+export interface BrowserHistoryDefinition {
+  navigate: NavigateFunction | unknown;
+  location: Location | unknown;
+}
+
+export const browserHistory: BrowserHistoryDefinition = {
```

Comment:
You can use an atom for this don't just store references globally they can create issues.